sphinx/search/__init__.py

```diff
@@ -250,6 +253,9 @@ class IndexBuilder:
         'pickle':   pickle
     }
 
+    _TRIE_CONTENTS = ""  # serializable sentinel value
```

Comment:
Nitpick: this value isn't serialized into the index file any longer, so refactoring to use a 'genuine' Python sentinel value could improve readability/Pythonicity here.

Reply:
Well, type-hinting support for sentinels seems limited (including in `typing_extensions`), so I'm going to pause on this.  There may be a clean/simple way to improve this and use a sentinel, but I'm not sure what that is yet.  Although slightly quirky, using the empty-string does keep the type signature of the trie node simple, and serializable (even if we perhaps don't need the latter).